Icinga should check whether the libsystemd library is available

We should update our cmake build files to check if libsystemd is available:

```
[ 27%] Building CXX object lib/base/CMakeFiles/base.dir/base_unity.cpp.o
cd /home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/lib/base && /usr/lib64/ccache/c++    -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9 -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/lib -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/third-party/execvpe -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/third-party/mmatch -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/third-party/socketpair  -O2 -g -m64 -fmessage-length=0 -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -pthread -Winvalid-pch -flto -std=c++0x -O2 -g -DNDEBUG   -o CMakeFiles/base.dir/application-version.cpp.o -c /home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/lib/base/application-version.cpp
cd /home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/lib/base && /usr/lib64/ccache/c++    -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9 -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/lib -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/third-party/execvpe -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/third-party/mmatch -I/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/third-party/socketpair  -O2 -g -m64 -fmessage-length=0 -D_FORTIFY_SOURCE=2 -fstack-protector -funwind-tables -fasynchronous-unwind-tables -g -pthread -Winvalid-pch -flto -std=c++0x -O2 -g -DNDEBUG   -o CMakeFiles/base.dir/base_unity.cpp.o -c /home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/lib/base/base_unity.cpp
In file included from /home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/lib/base/base_unity.cpp:1:0:
/home/jenkins/workspace/icinga2-snapshot/rpm-sles-12.2-1binary/arch/x86_64/rpmbuild/BUILD/icinga2-2.8.1.354.g5d464a9/lib/base/application.cpp:51:31: fatal error: systemd/sd-daemon.h: No such file or directory
 #include <systemd/sd-daemon.h>
                               ^
compilation terminated.
lib/base/CMakeFiles/base.dir/build.make:236: recipe for target 'lib/base/CMakeFiles/base.dir/base_unity.cpp.o' failed
make[2]: *** [lib/base/CMakeFiles/base.dir/base_unity.cpp.o] Error 1
```

--- a/INSTALL.md
+++ b/INSTALL.md
@@ -44,6 +44,10 @@ application using a dist tarball (including notes for distributions):
   - Alpine: boost-dev
 * GNU bison (bison)
 * GNU flex (flex) >= 2.5.35
+* Systemd headers
+  - Only required when using Systemd
+  - Debian/Ubuntu: libsystemd-dev
+  - RHEL/Fedora: systemd-devel
 
 ## Optional features
 

--- a/lib/base/CMakeLists.txt
+++ b/lib/base/CMakeLists.txt
@@ -97,6 +97,13 @@ if(ICINGA2_UNITY_BUILD)
   mkunity_target(base base base_SOURCES)
 endif()
 
+if(HAVE_SYSTEMD)
+  find_path(SYSTEMD_INCLUDE_DIR
+    NAMES systemd/sd-daemon.h
+    HINTS ${SYSTEMD_ROOT_DIR})
+  include_directories(${SYSTEMD_INCLUDE_DIR})
+endif()
+
 add_library(base OBJECT ${base_SOURCES})
 
 include_directories(${icinga2_SOURCE_DIR}/third-party/execvpe)

